doc/whats_new/upcoming_changes/sklearn.neighbors/26689.enhancement.rst

```diff
@@ -0,0 +1,9 @@
+- |Enhancement| Add :meth:`neighbors.NearestCentroid.decision_function`,
+  :meth:`neighbors.NearestCentroid.predict_proba` and
+  :meth:`neighbors.NearestCentroid.predict_log_proba`
+  to the :class:`neighbors.NearestCentroid` estimator class.
+  :pr:`26689` by :user:`Matthew Ning <NoPenguinsLand>`.
+
+- |Enhancement| Support the case when `X` is sparse and `shrinking_threshold`
+  is not `None` in :class:`neighbors.NearestCentroid`.
+  :pr:`26689` by :user:`Matthew Ning <NoPenguinsLand>`.
```

Comment:
```suggestion
- Add :meth:`neighbors.NearestCentroid.decision_function`,
  :meth:`neighbors.NearestCentroid.predict_proba` and
  :meth:`neighbors.NearestCentroid.predict_log_proba`
  to the :class:`neighbors.NearestCentroid` estimator class.
  By :user:`Matthew Ning <NoPenguinsLand>`

- Support the case when `X` is sparse and `shrinking_threshold`
  is not `None` in :class:`neighbors.NearestCentroid`.
  By :user:`Matthew Ning <NoPenguinsLand>`
```

@lesteve do we need two files here?

Reply:
I've seen this pattern often enough really. So ideally we'd be able to support this. However, in this case I don't mind it too much if the two points are merged. I do think having then separate is more natural though.

Reply:
There was nothing like this for `v1.6.rst` before we split it into fragments I think but it does seem to happen from time to time, according to a quick hacky `git grep` see below.

Personally I would advocate for doing the simplest thing with this towncrier thing ...

Having said this if you have multiple type or multiple sections, I think you can create multiple files. This would need to be tested to be 100% sure. This does seem to cover a good portion of the use cases I could look at (based on my hacky `git grep` thingy).

In this case they are all enhancement in the same `sklearn.neighbors` section so the filename would be named the same and grouping everything in a single bullet point seems OK enough for now.

As promised my hacky `git grep` thingy:
```bash
for f in doc/whats_new/v1*.rst; do
    echo $f
    git grep -P ':pr:`\d+`' $f | perl -pe 's@.*(:pr:`\d+`).*@\1@' | sort | uniq -c | sort -k 1 -n | grep -v '1 :pr:'
done  
```

Output:
```
doc/whats_new/v1.0.rst
      2 :pr:`19198`
      2 :pr:`19883`
      2 :pr:`20165`
      2 :pr:`20531`
      2 :pr:`20727`
      2 :pr:`21199`
      4 :pr:`17622`
doc/whats_new/v1.1.rst
      2 :pr:`10805`
      2 :pr:`20567`
      2 :pr:`21219`
      2 :pr:`21481`
      2 :pr:`21705`
      2 :pr:`21814`
      2 :pr:`22002`
      2 :pr:`22526`
      2 :pr:`23047`
      2 :pr:`23471`
doc/whats_new/v1.2.rst
      2 :pr:`22527`
      2 :pr:`22629`
      2 :pr:`23210`
      2 :pr:`23619`
      2 :pr:`23798`
      2 :pr:`23819`
      2 :pr:`23935`
      2 :pr:`24354`
      2 :pr:`24465`
      2 :pr:`24543`
      2 :pr:`25341`
      2 :pr:`25354`
      2 :pr:`25363`
      2 :pr:`25477`
doc/whats_new/v1.3.rst
      2 :pr:`24871`
      2 :pr:`25190`
      2 :pr:`25299`
      2 :pr:`25438`
      2 :pr:`25531`
      2 :pr:`25697`
      2 :pr:`25752`
      2 :pr:`26424`
      2 :pr:`26503`
      3 :pr:`25120`
      3 :pr:`26454`
      5 :pr:`26033`
doc/whats_new/v1.4.rst
      2 :pr:`13649`
      2 :pr:`26734`
      2 :pr:`26786`
      2 :pr:`26831`
      2 :pr:`26840`
      2 :pr:`27104`
      2 :pr:`27239`
      2 :pr:`27291`
      2 :pr:`27314`
      2 :pr:`27675`
      2 :pr:`27834`
      2 :pr:`28262`
      3 :pr:`26721`
      3 :pr:`27100`
      3 :pr:`27179`
      3 :pr:`27240`
      3 :pr:`27277`
      3 :pr:`27668`
      9 :pr:`27250`
doc/whats_new/v1.5.rst
      2 :pr:`18555`
      2 :pr:`27560`
      2 :pr:`27576`
      2 :pr:`28092`
      2 :pr:`29018`
      3 :pr:`27491`
```

Reply:
alright, so merge into one bullet point? @adrinjalali